### PR TITLE
8414 add records selection context inside the command menu

### LINKED ISSUE #8414: Add records selection context inside the command menu
## Scope & Context

Add records selection context inside the command menu.

image
image

## Current behavior

No context.

## Expected behavior

A record chip is displayed next to the command menu search bar when a record is selected.
When multiple records are selected, the number of selected records is displayed.
When all records are selected, 'All `objectName`' is displayed with the number of records.
When the user hits the `backspace` key when the search bar is empty, we should go back to the general command menu (ie. with no records selected)

## Technical inputs

Refactor `CommandMenu` by extracting the search bar into a component called `CommandMenuSearchBar`.
Create a component named `CommandMenuSearchBarContext`.
If the user hits `backspace` when the search bar is empty, we should reset the context of the command menu.

### PR DESCRIPTION
**Pull request by bosiraphael**
Closes #8414

https://github.com/user-attachments/assets/a6aeb50a-b57d-43db-a839-4627c49b4155

### CODE REVIEW

**Comment by Bonapara** [comment]
Great! Some feedbacks:

### Dark Mode

CleanShot 2024-11-20 at 18 40 48


### Need two escape to remove the selection

When focus is on "Type anything" first escape remove the focus on the input, second escape remove the selection

### Removing the selection in breadcrumb should remove the selection on table/kanbans

If i press ⌫ or escape, should also deselect the record on the left

### Header size

Should be 56px to match. Check all the header paddings

CleanShot 2024-11-20 at 18 47 48

### Chip should be 32px heigh

put the border inside or make the chip 30px if the border are outside

CleanShot 2024-11-20 at 18 50 05

### Delete doesn't work

https://github.com/user-attachments/assets/594a3379-0f19-4d16-bd1d-39b291857bef

### Icon identifiers

People icon identifiers should be rounded

CleanShot 2024-11-20 at 18 52 30

Tasks & Notes should use their colored icon image identifiers

CleanShot 2024-11-20 at 18 53 08

### Empty state in wrongly aligned

CleanShot 2024-11-20 at 18 54 31

File: packages/twenty-front/src/modules/command-menu/components/CommandMenuContextRecordChip.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const StyledAvatarContainer = styled.div`
+  display: flex;
+`;
+
+const CommandMenuContextRecordChipAvatars = ({
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/command-menu/components/CommandMenuContextRecordChip.tsx:48]: 1 type per file, 1 component per file

Comment (charlesBochet) [reply packages/twenty-front/src/modules/command-menu/components/CommandMenuContextRecordChip.tsx:48]: add stories

File: packages/twenty-front/src/modules/context-store/components/MainContextStoreComponentInstanceIdSetterEffect.tsx [hunk 2]
```diff
@@ -11,10 +11,10 @@ export const MainContextStoreComponentInstanceIdSetterEffect = () => {
   const context = useContext(ContextStoreComponentInstanceContext);
 
   useEffect(() => {
-    setMainContextStoreComponentInstanceId(context?.instanceId ?? null);
+    setMainContextStoreComponentInstanceId(context?.instanceId ?? 'app');
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/context-store/components/MainContextStoreComponentInstanceIdSetterEffect.tsx:14]: make a constant

File: packages/twenty-front/src/modules/context-store/hooks/useContextStoreCurrentObjectMetadataIdOrThrow.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { contextStoreCurrentObjectMetadataIdComponentState } from '@/context-store/states/contextStoreCurrentObjectMetadataIdComponentState';
+import { useRecoilComponentValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValueV2';
+
+export const useContextStoreCurrentObjectMetadataIdOrThrow = (
+  instanceId?: string,
+) => {
+  const contextStoreCurrentObjectMetadataIdComponent =
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/context-store/hooks/useContextStoreCurrentObjectMetadataIdOrThrow.ts:7]: naming

File: packages/twenty-front/src/modules/context-store/hooks/useContextStoreSelectedRecords.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import { useContextStoreCurrentObjectMetadataIdOrThrow } from '@/context-store/hooks/useContextStoreCurrentObjectMetadataIdOrThrow';
+import { contextStoreFiltersComponentState } from '@/context-store/states/contextStoreFiltersComponentState';
+import { contextStoreTargetedRecordsRuleComponentState } from '@/context-store/states/contextStoreTargetedRecordsRuleComponentState';
+import { computeContextStoreFilters } from '@/context-store/utils/computeContextStoreFilters';
+import { useObjectMetadataItemById } from '@/object-metadata/hooks/useObjectMetadataItemById';
+import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
+import { useRecoilComponentValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValueV2';
+
+export const useContextStoreSelectedRecords = ({
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/context-store/hooks/useContextStoreSelectedRecords.ts:9]: useFindManyRecordSelectedInContextStore

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**